com/craftinginterpreters/lox/LoxClass.java

```diff
@@ -3,11 +3,14 @@
 import java.util.List;
 import java.util.Map;
 
-class LoxClass implements LoxCallable {
+class LoxClass extends LoxInstance implements LoxCallable {
     final String name;
     private final Map<String, LoxFunction> methods;
 
     LoxClass(String name, Map<String, LoxFunction> methods) {
+        // I need a class for the LoxInstance constructor
+        // That's going to be the metaclass for this class
+        super(new LoxClass(name + "Metaclass")); // This would end up being an infinite loop
```

Comment:
I need some sort of singleton `Metaclass` that will terminate the hierarchy here.